### PR TITLE
Wait for controllerrevision informer to sync on statefulset controller startup

### PR DESCRIPTION
**Pull request by tnozicka**
Fixes: https://github.com/kubernetes/kubernetes/issues/56355

Needs to be backported to at least 1.8, 1.9 and triaged for impact on 1.9 release

/cc @smarterclayton @kow3ns @enisoc 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Review by kow3ns — APPROVED** [review]

**Comment by smarterclayton** [comment]
Serious issue for stateful set users and can result in large numbers of controller revisions created unexpectedly.  Minor risk.

**Comment by tnozicka** [comment]
status: needs lgtm :)

**Comment by kow3ns** [comment]
@enisoc @foxish PTAL

**Comment by kow3ns** [comment]
@tnozicka I like to get a second review as apposed to lgtm and approve from the same person.

**Review by janetkuo — APPROVED** [review]

**Comment by krmayankk** [comment]
@tnozicka @kow3ns @smarterclayton can this be also back ported to 1.7 ? We are seeing this issue as well and we just upgraded to 1.7. It would be nice to get this fix without upgrading to 1.8 or 1.9 . Also are there ways to avoid it ?

**Comment by tnozicka** [comment]
@krmayankk syncing informers should be there from the start so I would say backport to 1.7 should be ok but the likely offending PR with collision count https://github.com/kubernetes/kubernetes/pull/50490 isn't part of 1.7 I think. Without collision count it should fail on the controller revision already being created. It might be caused by something similar though.

**Pull request merged**
